### PR TITLE
Clean title output

### PR DESCRIPTION
**Pull request by domoritz**

### CODE REVIEW

**Comment by domoritz** [comment]
Ready for review @kanitw

File: src/compile/axis/parse.ts [hunk 1]
```diff
@@ -317,7 +317,8 @@ function getProperty<K extends keyof VgAxis>(property: K, specifiedAxis: Axis, c
     case 'ticks':
       return properties.ticks('ticks', specifiedAxis, isGridAxis, channel);
     case 'title':
-      return getSpecifiedOrDefaultValue(specifiedAxis.title, properties.title(specifiedAxis.titleMaxLength, fieldDef, model.config));
+      const title = getSpecifiedOrDefaultValue(specifiedAxis.title, properties.title(specifiedAxis.titleMaxLength, fieldDef, model.config));
+      return title === null || title === '' ? undefined : title;
```
Comment (kanitw) [inline src/compile/axis/parse.ts]: what if title is `undefined`? 

Why not just 
```
return getSpecifiedOrDefaultValue(specifiedAxis.title, properties.title(specifiedAxis.titleMaxLength, fieldDef, model.config)) || undefined;
```

File: src/compile/legend/parse.ts [hunk 2]
```diff
@@ -96,7 +96,8 @@ function getProperty(property: keyof (Legend | VgLegend), specifiedLegend: Legen
       // We don't include temporal field here as we apply format in encode block
       return numberFormat(fieldDef, specifiedLegend.format, model.config);
     case 'title':
-      return getSpecifiedOrDefaultValue(specifiedLegend.title, fieldDefTitle(fieldDef, model.config));
+      const title = getSpecifiedOrDefaultValue(specifiedLegend.title, fieldDefTitle(fieldDef, model.config));
+      return title === null || title === '' ? undefined : title;
```
Comment (kanitw) [inline src/compile/legend/parse.ts]: Why not just 
```
return getSpecifiedOrDefaultValue(specifiedAxis.title, properties.title(specifiedAxis.titleMaxLength, fieldDef, model.config)) || undefined;
```

File: src/compile/axis/parse.ts (hunk 1 above)
Comment (domoritz) [reply src/compile/axis/parse.ts]: Then we want to use the default, right?

Comment (kanitw) [reply src/compile/axis/parse.ts]: I see, but then title is never undefined here I guess. 

Still, I think just do `|| undefined` makes it less confusing.

Comment (domoritz) [reply src/compile/axis/parse.ts]: I agree

**Pull request merged**